asserts/snapasserts/validation_sets.go

```diff
@@ -1031,33 +990,137 @@ func (v *ValidationSets) SnapConstrained(snapRef naming.SnapRef) bool {
 	return v.constraintsForSnap(snapRef) != nil
 }
 
-// CheckPresenceInvalid returns the list of all validation sets that declare
-// presence of the given snap as invalid. PresenceConstraintError is returned if
-// presence of the snap is "optional" or "required".
-// The method assumes that validation sets are not in conflict.
-func (v *ValidationSets) CheckPresenceInvalid(snapRef naming.SnapRef) ([]ValidationSetKey, error) {
-	cstrs := v.constraintsForSnap(snapRef)
+type presence struct {
```

Comment:
wouldn't it be a bit clearer if this was exported?

Reply:
Yeah, probably. I was having a hard time naming it, since Presence technically already has a meaning.